aten/src/ATen/templates/Tensor.h

```diff
@@ -120,6 +126,33 @@ struct Tensor : public detail::TensorBase {
   Tensor operator[](Tensor index) const;
   Tensor operator[](int64_t index) const;
 
+  // ~~~~~ Autograd API ~~~~~
+
+  void set_requires_grad(bool requires_grad) {
+    pImpl->set_requires_grad(requires_grad);
+  }
+  bool requires_grad() const noexcept {
+    return pImpl->requires_grad();
+  }
+
+  Tensor& grad() noexcept {
```

Comment:
I would much prefer a `set_grad()` virtual method, as opposed to a method that returns a reference that can be modified.

Reply:
Also, all of this const business isn't really getting anything useful done, c.f. https://github.com/zdevito/ATen/issues/27

Reply:
I think it's still useful once you downcast the `Tensor&` to `Variable&`, since `const` is useful for `Variable`